sensor/common/scan/scan.go

```diff
@@ -163,57 +131,171 @@ func (s *LocalScan) enrichLocalImageFromRegistry(ctx context.Context, centralCli
-	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, image)
+	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, pullSourceImage)
 
 	// Fetch signatures associated with image from registry.
-	sigs := s.fetchSignatures(ctx, errorList, reg, image)
+	sigs := s.fetchSignatures(ctx, errorList, reg, pullSourceImage)
 
 	// Send local enriched data to central to receive a fully enrich image. This includes image vulnerabilities and
 	// signature verification results.
 	centralResp, err := centralClient.EnrichLocalImageInternal(ctx, &v1.EnrichLocalImageInternalRequest{
-		ImageId:        image.GetId(),
-		ImageName:      image.GetName(),
-		Metadata:       image.GetMetadata(),
+		ImageId:        srcImage.GetId(),
+		ImageName:      srcImage.GetName(),
+		Metadata:       pullSourceImage.GetMetadata(),
 		Components:     scannerResp.GetComponents(),
 		Notes:          scannerResp.GetNotes(),
 		ImageSignature: &storage.ImageSignature{Signatures: sigs},
-		ImageNotes:     image.GetNotes(),
+		ImageNotes:     pullSourceImage.GetNotes(),
 		Error:          errorList.String(),
 		RequestId:      requestID,
 		Force:          force,
 	})
 	if err != nil {
-		log.Debugf("Unable to enrich image %q: %v", image.GetName().GetFullName(), err)
-		return nil, pkgErrors.Wrapf(err, "enriching image %q via central", image.GetName())
+		log.Debugf("Unable to enrich image %q: %v", srcImage.GetName().GetFullName(), err)
+		return nil, pkgErrors.Wrapf(err, "enriching image %q via central", srcImage.GetName())
 	}
 
 	if errorList.Empty() {
-		log.Debugf("Retrieved image enrichment results for %q with id %q", image.GetName().GetFullName(), image.GetId())
+		log.Debugf("Retrieved image enrichment results from Central for %q with id %q (%v) components", srcImage.GetName().GetFullName(), srcImage.GetId(), centralResp.GetImage().GetComponents())
 	}
 
 	return centralResp.GetImage(), errorList.ToError()
 }
 
+// getImageWithMetadata on success returns the registry used to pull metadata and an image with metadata populated.
+// The image returned may represent the source image or an image from a registry mirror.
+func (s *LocalScan) getImageWithMetadata(ctx context.Context, errorList *errorhelpers.ErrorList, namespace string, sourceImage *storage.ContainerImage) (registryTypes.ImageRegistry, *storage.Image) {
+	// Obtain the pull sources, which will include mirrors.
+	pullSources := s.getPullSources(sourceImage)
+
+	// errs are only added to errorList when attempts from all pull sources + all registries fail.
+	errs := errorhelpers.NewErrorList("")
+
+	// For each pull source, obtain the associated registries and attempt to obtain metadata, stopping on first success.
+	for _, pullSource := range pullSources {
+		registries, err := s.getRegistries(ctx, namespace, pullSource.GetName())
+		if err != nil {
+			log.Warnf("Error getting registries for pull source %q, skipping: %v", pullSource.GetName().GetFullName(), err)
```

Comment:
Minimal noisiness (1/10 noisiness, is there a noisiness scale? j/k) - `getRegistries` will only return an error if:

No secrets were found amongst sync'd image integrations AND 
no secrets found from namespace secrets AND 
no match in the global pull secret AND 
the TLS check fails when creating a no-auth registry AND 
the TLS check failed with an error that isn't already ignored / handled (such as an x509 issue does not return an error).

I don't believe these will all be true often which is why I don't believe this will be noisy.  Happy to change this to Debug if feel differently.